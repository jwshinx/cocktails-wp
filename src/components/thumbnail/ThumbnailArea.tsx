import React from 'react'

import { GeneralDrink } from '../../models/cocktail'

export default function ThumbnailArea({ item }: { item: GeneralDrink }) {
  return (
    <>
      <img
        className="card-img-top"
        src={item.strDrinkThumb}
        alt={item.strDrink}
      />
    </>
  )
}
