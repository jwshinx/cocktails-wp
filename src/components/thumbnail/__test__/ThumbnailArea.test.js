/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

import ThumbnailArea from '../ThumbnailArea'

const data = {
  strDrink: 'Sidecar',
  idDrink: '45',
  strDrinkThumb: 'http:some-url',
}

it('should render an image with alt value "Sidecar"', () => {
  render(<ThumbnailArea item={data} />)
  const elem = screen.getByAltText(/Sidecar/)
  expect(elem).toBeInTheDocument()
})

it('should snapshot ThumbnailArea', () => {
  const tree = renderer.create(<ThumbnailArea item={data} />).toJSON()
  expect(tree).toMatchSnapshot()
})
