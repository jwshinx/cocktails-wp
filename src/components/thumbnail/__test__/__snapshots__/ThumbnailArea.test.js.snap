// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot ThumbnailArea 1`] = `
<img
  alt="Sidecar"
  className="card-img-top"
  src="http:some-url"
/>
`;
